src/generators/angular/AngularServicesMethodGenerator.ts

```diff
@@ -157,10 +168,8 @@ export class AngularServicesMethodGenerator {
             ? `Promise<${x.returnType?.type.type}>`
             : `Observable<${this.getReturnTypeName(x.returnType, x.returnType?.type.type)}>`;
     }
-    private createDownloadMethod(writer: CodeBlockWriter, model: IMethodModel): void {
-        const bodyParameters = model.parameters.filter((z) => z.name !== 'saveAs' && z.place === ParameterPlace.Body);
-        const dataParameter = bodyParameters.find((z) => z.name === 'data');
-        const optionsParameter = bodyParameters.find((z) => z.name === 'options');
+    private createDownloadMethod(writer: CodeBlockWriter, model: IMethodModel, options?: Partial<MethodParameter>): void {
+        const parameter = first(model.parameters.filter((z) => z.name !== 'saveAs' && z.place === ParameterPlace.Body));
```

Comment:
we do not use anything from MethodParameter except for the name
maybe type should be optionsParameter?: { name: string }